packages/cli/test/unit/ActiveExecutions.test.ts

```diff
@@ -0,0 +1,92 @@
+import { ActiveExecutions, IWorkflowExecutionDataProcess, Db } from '../../src';
+import { mocked } from 'jest-mock';
+import PCancelable from 'p-cancelable';
+import { IRun } from 'n8n-workflow';
+
+const FAKE_EXECUTION_ID = '15';
+const FAKE_SECOND_EXECUTION_ID = '20';
+
+jest.mock('../../src/Db', () => {
+	return {
+		collections: {
+			Execution: {
+				save: jest.fn(async () => Promise.resolve({id: FAKE_EXECUTION_ID})),
+				update: jest.fn(),
+			}
+		}
+	};
+});
+
+describe('ActiveExecutions', () => {
+
+	let activeExecutions: ActiveExecutions.ActiveExecutions;
+
+	beforeEach(() => {
+		activeExecutions = new ActiveExecutions.ActiveExecutions();
+	});
+
+	afterEach(() => {
+		jest.clearAllMocks();
+	});
+
+	test('Should initialize activeExecutions with empty list', () => {
+		expect(activeExecutions.getActiveExecutions().length).toBe(0);
+	});
+
+	test('Should add execution to active execution list', async () => {
+		const newExecution = generateFakeWorkflowExecutionDataProcess();
+		const executionId = await activeExecutions.add(newExecution);
+		expect(executionId).toBe(FAKE_EXECUTION_ID);
+		expect(activeExecutions.getActiveExecutions().length).toBe(1);
+		expect(mocked(Db.collections.Execution.save)).toHaveBeenCalledTimes(1);
+		expect(mocked(Db.collections.Execution.update)).toHaveBeenCalledTimes(0);
+	});
```

Comment:
For learning, should we be making expectations about internal `Db` calls? I would have treated this as implementation detail, using `testDb` instead of mocking.

Reply:
We could use `testDb` but that's more useful if I want to write and maybe read afterwards. In this case I just want to make sure that we're making the call since the actual result doesn't matter.

Considering the (minimal) overhead, it's worth not writing anything to disk since it could add up over time.

Reply:
> I just want to make sure that we're making the call 

Asking only in principle, to write better tests in future - in cases like this why do we want to ensure that these DB calls are made? My reasoning was that an implementation detail should not be tested, so that in future refactoring the implementation without changing the internal state will not require us to come back to adjust the tests.

Reply:
In fact I am making this assertion because I want to assert that the business logic is being followed, such as creating a new object vs updating an existing. Also if we change the underlying implementation, as long as we continue using te same `Db.collections.Executions.<something>` the test should continue to work, but if we change it, the tests should also reflect that this business logic (saving vs updating) needs to be maintained.